src/ring_theory/dedekind_domain/adic_valuation.lean

```diff
@@ -333,3 +333,43 @@ instance adic_completion.has_lift_t : has_lift_t K (v.adic_completion K) :=
 def adic_completion_integers : valuation_subring (v.adic_completion K) := valued.v.valuation_subring
 
 end is_dedekind_domain.height_one_spectrum
+
+variables (R K)
+
+lemma adic_completion.is_integer (x : v.adic_completion K) :
+  x ∈ v.adic_completion_integers K ↔ (valued.v x : with_zero (multiplicative ℤ)) ≤ 1 := by refl
+
+/-- The natural inclusion of `R` in `adic_completion`. -/
+def inj_adic_completion_integers' : R → (v.adic_completion K) :=
+λ r, (coe : K → (v.adic_completion K)) (algebra_map R K r)
+
+/-- The natural inclusion of `R` in `adic_completion_integers`. -/
+def inj_adic_completion_integers : R → (v.adic_completion_integers K) :=
+λ r, ⟨(coe : K → (v.adic_completion K)) (algebra_map R K r), begin
+  change @valued.extension K _ _ _ v.adic_valued (algebra_map R K r) ≤ 1,
+  rw @valued.extension_extends K _ _ _ v.adic_valued (algebra_map R K r),
+  exact v.valuation_le_one _,
+end⟩
+
+lemma inj_adic_completion_integers.injective :
+  function.injective (inj_adic_completion_integers R K v) :=
+begin
+  intros x y hxy,
+  have h_inj : function.injective (coe : K → v.adic_completion K) :=
+    @uniform_space.completion.coe_injective K v.adic_valued.to_uniform_space _,
+  rw [inj_adic_completion_integers, subtype.mk_eq_mk] at hxy,
+  exact (is_fraction_ring.injective R K) ((h_inj) hxy)
```

Comment:
I think it's worth applying `subtype.ext_iff` to `hxy` here.

For example the following doesn't require unfolding the definition of `inj_adic_completion_integers`:
```suggestion
  replace hxy : (↑(algebra_map R K x) : v.adic_completion K) = ↑(algebra_map R K y),
  { simpa using subtype.ext_iff.mp hxy, },
  exact is_fraction_ring.injective R K (h_inj hxy),
```

Reply:
Does this work fast on your computer? In my laptop, the proof unfolding `inj_adic_completion_integers` takes 174ms, but the proof using `simpa` takes 2.79s. I haven't changed it for this reason.

Reply:
OK no need to use `simpa`, my point was that I think it's better to use `subtype.ext_iff` (partly since then we can avoid unfolding `inj_adic_completion_integers`).

I'll make this a suggestion in a separate comment since this comment is attached to an outdated commit.